index.bs

```diff
@@ -0,0 +1,221 @@
+
+<dfn attribute for=PerformanceLongTaskTiming>attribution</dfn>
+A user agent implementing {{LayoutShift}} must perform the following steps:
+1. Run the <a>register a performance entry type</a> algorithm with <code>"layoustShift"</code> as input.
+
+
+NOTE: The <a>getEntriesByType</a> method will not return entries of this type.  To obtain instability events that occur before a <a>PerformanceObserver</a> has been registered, register the observer with <a>buffered</a> set to true.
+
+
+
+
+Processing model {#sec-processing-model}
+========================================
+
+
+Modifications to the HTML specification {#sec-modifications-HTML}
+--------------------------------------------------------
+
+<em>This section will be removed once the <a href=https://html.spec.whatwg.org/multipage>HTML specification</a> has been modified.</em>
+
+For each <a>element</a>, there is a:
+* <dfn>currentFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>previousFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>currentVisualRepresentation</dfn> which is initially set to <code>null</code>.
+* <dfn>previousVisualRepresentation</dfn> with is initially set to <code>null</code>.
+
+<div algorithm="additions to update rendering">
+    In the <a>update the rendering</a> step of the <a>event loop processing model</a>, add a step right after the step that calls <a>mark paint timing</a>:
+
+    1. Invoke the algorithm to <a>evaluate the layout instability</a> of a document with the complete set of <a>element</a>s for that {{Document}}.
+</div>
+
+
+Evaluate Layout Instability {#sec-eval-layout-instability}
+--------------------------------------------------------
+
+<div algorithm="evaluate the layout instability">
+    When asked to <dfn export>evaluate the layout instability</dfn>, with <var>elements</var> as the list of elements to be evaluated, run the following steps:
+
+    1. Let <var>unstable elements</var> be an empty list.
+    1. Let <var>total width</var> be the current <a>viewport</a> width.
+    1. Let <var>total height</var> be the current <a>viewport</a> height. 
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <a>currentFrameStartingPoint</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <a>currentVisualRepresentation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>currentFrameStartingPoint</a> and <a>previousFrameStartingPoint</a>.  
+        1. If <var>unstable</var> is true, add a tuple to <var>unstable elements</var> with:
+            1. <a>currentVisualRepresentation</a> of <var>element</var>
+            1. <a>previousVisualRepresentation</a> of <var>element</var>
+            1. <var ignore=''>shiftFraction</var> of <var>element</var>:
```

Comment:
Nit: not clear what 'shiftFraction' is meant to achieve here. Maybe first compute the shiftFraction variable, and then add it to the tuple.

Reply:
Done.